server.go

```diff
@@ -1103,7 +1107,13 @@ func (sp *serverPeer) OnInv(p *peer.Peer, msg *wire.MsgInv) {
 
 // OnHeaders is invoked when a peer receives a headers wire message.  The
 // message is passed down to the net sync manager.
-func (sp *serverPeer) OnHeaders(_ *peer.Peer, msg *wire.MsgHeaders) {
+func (sp *serverPeer) OnHeaders(p *peer.Peer, msg *wire.MsgHeaders) {
+	// Ban peers sending empty headers requests.
+	if len(msg.Headers) == 0 {
```

Comment:
Let's revert this change for this PR.  I'm fairly certain it is possible for an empty headers message to be sent in valid circumstances when reaching the chain tip during sync and this would end up banning them due to no fault of their own.